cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -147,6 +147,30 @@ public void setSnapshot(T group, Snapshot snapshot) {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override public Snapshot getSnapshot(T group) {
+    readLock.lock();
+
+    try {
+      Snapshot snapshot = snapshots.get(group);
```

Comment:
The fields in `Snapshot` are all intentionally immutable, so we should be able to skip the copy and just return `snapshots.get(group)` directly.